actions/test/run_local.ts

```diff
@@ -111,6 +109,23 @@ async function processBlock(
         });
     }
   }
+
+  // Block watcher for creating new orders
+  const testBlockEvent = new TestBlockEvent();
+  testBlockEvent.blockNumber = blockNumber;
+  testBlockEvent.blockDifficulty = block.difficulty.toString();
+  testBlockEvent.blockHash = block.hash;
+  testBlockEvent.network = chainId.toString();
+
+  // run action
+  await testRuntime
+    .execute(checkForAndPlaceOrder, testBlockEvent)
+    .catch((e) => {
+      console.log(
+        "[run_local] Error in checkForAndPlaceOrder processing block",
+        e
+      );
+    });
```

Comment:
I moved the order of the execution of the actions.

Order matters here, because `addContract` will register the order, and `checkForAndPlaceOrder` will find the order persisted and will create it in the orderbook. 

This part was making the script to add the order in the orderbook for NEXT BLOCK not in the current one where it was defined.

Reply:
Nice catch! Yeah, this is an obvious one :see_no_evil: my bad.